### ISSUE TITLE
[Bug]: @storybook/core is (wrongly) specifying @storybook/global as a devDependency

### ISSUE THREAD

**Issue by alexanderbartl** [storybookjs/storybook#28619]
### Describe the bug

Storybook 8.2 (re-)introduces the `@storybook/core` dependency. That package imports from `@storybook/global`, but specifies it only as a `devDependency`. This works just fine in most setups because various other storybook packages do specify `@storybook/global` as a dependency, so it gets installed anyway.

However, when using yarn berry with [Plug'n'Play](https://yarnpkg.com/features/pnp), a package can only access other packages if they are specified in the dependency list.
There's a workaround using [packageExtensions](https://yarnpkg.com/configuration/yarnrc#packageExtensions) in the `.yarnrc.yml`, but IMO the proper solution would be to upgrade `@storybook/global` to a proper dependency of `@storybook/core`.

### Reproduction link

probably not necessary for this case

### System

```bash
Shouldn't be relevant to the problem at hand, can be provided if necessary.
```

**Comment by shilman** [comment]
@alexanderbartl what's the actual bug you're hitting? we're using `tsup` so when a package is listed as a dev dep, it actually gets bundled into the dist bundle if it's referred to by code inside the package. so dev dep probably doesn't mean what you think it means in this context and is intentional. if you share the problem you're hitting we can go from there.

**Comment by alexanderbartl** [comment]
@shilman Storybook itself seems to run fine. When running tests with jest (and using `composeStories`), I get the following error message:

    Cannot find module '@storybook/global' from '/home/alex/local/path/to/repo/.yarn/cache/@storybook-core-npm-8.2.0-bd0bae05cc-bdc4f37182.zip/node_modules/@storybook/core/dist/components'

Adding the following to the `.yarnrc.yml` solves the issue:

     packageExtensions:
       "@storybook/core@*":
          dependencies:
            "@storybook/global": "*"

So does disabling Plug'n'Play.

I haven't used `tsup` yet, so I'm not sure what to expect. What I can tell is that I'm seeing imports from `@storybook/global` in `dist/components/index.js` and `dist/components.index.cjs`.

**Comment by shilman** [comment]
@alexanderbartl Thanks for the clarification. @yannbf @kasperpeulen any thoughts on portable stories or do we need a repro?

**Comment by kasperpeulen** [comment]
Found the issue with @ndelangen will make a PR

**Comment by alexanderbartl** [comment]
Thanks for looking into it, @kasperpeulen and @ndelangen! I can confirm that the issue is fixed in `8.3.0-alpha.2`.
